pyomo/gdp/plugins/chull.py

```diff
@@ -210,231 +233,205 @@ def _apply_to_impl(self, instance, **kwds):
                                 % (t.name, instance.name))
             elif t.ctype is Disjunction:
                 if t.parent_component() is t:
-                    self._transformDisjunction(t, transBlock)
+                    self._transform_disjunction(t)
                 else:
-                    self._transformDisjunctionData(t, transBlock, t.index())
+                    self._transform_disjunctionData(t, t.index())
             elif t.ctype in (Block, Disjunct):
                 if t.parent_component() is t:
-                    self._transformBlock(t, transBlock)
+                    self._transform_block(t)
                 else:
-                    self._transformBlockData(t, transBlock)
+                    self._transform_blockData(t)
```

Comment:
An alternative test to this is:
```python
elif t.ctype is Disjunction:
    if t.is_indexed():
        self._transform_disjunction(t)
    else:
        self._transform_disjunctionData(t, t.index())
elif t.ctype in (Block, Disjunct):
    if t.is_indexed():
        self._transform_block(t)
    else:
        self._transform_blockData(t)
```
...except that SimpleDisjunctions will now be processed by `_transform_disjunctionData` instead of `_transform_disjunction` (and the same for SimpleBlock and SimpleDisjunct)

Reply:
Oh, nice, thank you! There's no problem with that change, it makes the code make a lot more sense, actually.